pulp_ansible/app/viewsets.py

```diff
@@ -327,6 +329,30 @@ def sync(self, request, pk):
         )
         return OperationPostponedResponse(result, request)
 
+    @extend_schema(
+        description="Trigger an asynchronous task to rebuild Ansible content meta.",
+        responses={202: AsyncOperationResponseSerializer},
+    )
+    @action(detail=True, methods=["post"], serializer_class=AnsibleRepositoryRebuildSerializer)
+    def rebuild_metadata(self, request, pk):
+        """
+        Dispatches a collection version rebuild task.
+        """
+        repository = self.get_object()
+        # serializer = AnsibleRepositoryRebuildSerializer(
+        #     data=request.data, context={"request": request, "repository_pk": repository.pk}
+        # )
+        # serializer.is_valid(raise_exception=True)
+
+        result = dispatch(
+            rebuild_repository_collection_versions_metadata,
+            exclusive_resources=[repository],
```

Comment:
Probably safe to put repository lock in `shared_resources`. Question is should this also have an exclusive lock on all of `/pulp/api/v3/content/ansible/collection_versions/` since it's modifying the collections. It might be useless since I don't think there are any other tasks that require that lock even if they create or read collections.

Reply:
I see this as repairing and not changing, and also yes we don't lock this in other places.